packages/ragbits-core/src/ragbits/core/vector_stores/_cli.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+
+from ragbits.cli import cli_state, print_output
+from ragbits.cli.state import OutputType
+from ragbits.core.config import core_config
+from ragbits.core.embeddings.base import Embeddings
+from ragbits.core.utils.config_handling import InvalidConfigError
+from ragbits.core.vector_stores.base import VectorStore, VectorStoreOptions
+
+vector_stores_app = typer.Typer(no_args_is_help=True)
+
+
+@dataclass
+class CLIState:
+    vector_store: VectorStore | None = None
+
+
+state: CLIState = CLIState()
+
+
+@vector_stores_app.callback()
+def common_args(
```

Comment:
if we keep vector store config instead of vector store in the cli state - we could use this callback to eventually override the config - with no need to use it whenever we want to use any command

Reply:
I don't understand. Just like described in the ticket for this feature, you can use one of four ways to choose the store that you want to create:
1. Use the default store configuration from the project's `pyproject.yaml` (in this case ragbits will automatically create a store object from this configuration for you)
2. Use the default store factory function from the project's `pyproject.yaml` (in this case ragbits will automatically create a store object by calling the factory function)
3. Explicitly pass the filesystem path to a file containing store configuration (in this case ragbits will create a store object from this configuration)
4. Explicitly pass the Python path to a factory function (in this case ragbits will create a store object by calling the factory function).

This is the behavior that we agreed on during grooming. There is "no need to use it whenever we want to use any command" - the `factory_path` and `yaml_path` on the callback are optional and are used only when you want to override the defaults from the project's configuration for the given run of the command.

Reply:
ok - i forgot that callback is always run eventually with default arguments and had a misconception that if we want to run it we need to pass the arguments

Reply:
More precisely callback can be run with empty arguments and the `subclass_from_defaults` method used inside of it is responsible for using the default in such case.